engine/src/main/java/org/terasology/config/flexible/Setting.java

```diff
@@ -0,0 +1,124 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible;
+
+import com.google.common.collect.Lists;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.config.flexible.validators.SettingValueValidator;
+import org.terasology.utilities.subscribables.GeneralSubscribable;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.List;
+
+public class Setting<T> implements GeneralSubscribable {
+    private final T defaultValue;
+
+    private ResourceUrn id;
+
+    private String idString;
+    private T value;
+
+    private String name;
+
+    private String description;
+    private SettingValueValidator<T> valueValidator;
+    private List<PropertyChangeListener> subscribers;
+
+    public Setting(ResourceUrn id, T defaultValue, SettingValueValidator<T> valueValidator) {
+        this.id = id;
+        // Precomputing for performance
+        this.idString = id.toString();
+
+        this.defaultValue = defaultValue;
+        this.value = this.defaultValue;
+
+        this.valueValidator = valueValidator;
+
+        this.subscribers = Lists.newArrayList();
+    }
+
+    public Class<? extends SettingValueValidator> getValidatorClass() {
+        return valueValidator.getClass();
+    }
+
+    public SettingValueValidator<T> getValueValidator() {
+        return valueValidator;
+    }
+
+    public void setValueValidator(SettingValueValidator<T> valueValidator) {
+        this.valueValidator = valueValidator;
+    }
+
+    private void dispatchChangedEvent(PropertyChangeEvent event) {
+        for (PropertyChangeListener subscriber : subscribers) {
+            subscriber.propertyChange(event);
+        }
+    }
+
+    public void subscribe(PropertyChangeListener listener) {
+        subscribers.add(listener);
+    }
+
+    public void unsubscribe(PropertyChangeListener listener) {
+        subscribers.remove(listener);
+    }
+
+    public boolean hasSubscribers() {
+        return !subscribers.isEmpty();
+    }
+
+    public ResourceUrn getId() {
+        return id;
+    }
+
+    public T getDefaultValue() {
+        return defaultValue;
+    }
+
+    public T getValue() {
+        return value;
+    }
+
+    public boolean setValue(T value) {
+        if (!valueValidator.isValid(value))
+            return false;
+
+        PropertyChangeEvent event = new PropertyChangeEvent(this, idString, this.value, value);
+
+        this.value = value;
+
+        dispatchChangedEvent(event);
+
+        return true;
+    }
+
+    public String getName() {
+        return name;
+    }
```

Comment:
We don't have a clear strategy on how to expand internationalization coverage beyond the main (static) menu. I would just make sure we have clear non-ambiguous keys, maybe prefixed with something to tie them to their origin config, so we can associate that with a translation entry at some point later.